app/Controllers/updateController.php

```diff
@@ -11,31 +11,54 @@ public static function hasGitUpdate() {
 		chdir(FRESHRSS_PATH);
 		$output = array();
 		try {
-			exec('git fetch', $output, $return);
+			exec('git fetch --prune', $output, $return);
 			if ($return == 0) {
+				unset($output);
 				exec('git status -sb --porcelain remote', $output, $return);
 			} else {
-				$line = is_array($output) ? implode('; ', $output) : '' . $output;
+				$line = is_array($output) ? implode('; ', $output) : $output;
 				Minz_Log::warning('git fetch warning:' . $line);
 			}
 		} catch (Exception $e) {
 			Minz_Log::warning('git fetch error:' . $e->getMessage());
 		}
 		chdir($cwd);
-		$line = is_array($output) ? implode('; ', $output) : '' . $output;
-		return strpos($line, '[behind') !== false || strpos($line, '[ahead') !== false;
+		$line = is_array($output) ? implode('; ', $output) : $output;
+		return strpos($line, '[behind') !== false || strpos($line, '[ahead') !== false || strpos($line, '[gone') !== false;
 	}
 
 	public static function gitPull() {
 		$cwd = getcwd();
 		chdir(FRESHRSS_PATH);
-		$output = '';
+		$output = [];
 		$return = 1;
 		try {
-			exec('git fetch', $output, $return);
+			exec('git fetch --prune', $output, $return);
 			if ($return == 0) {
+				unset($output);
 				exec('git reset --hard FETCH_HEAD', $output, $return);
 			}
+
+			// Automatic change to the new name of edge branch since FreshRSS 1.18.0
```

Comment:
This can be put in a [migration](https://github.com/FreshRSS/FreshRSS/blob/edge/docs/en/developers/Minz/migrations.md) instead to avoid more complicated code here

Reply:
Otherwise, it worked for me :+1:

Reply:
Ah yes, I am not familiar with this part of the code yet...

Reply:
Something like that should work (not tested):

```php
// File: app/migrations/2021_04_24_GitCheckoutFromMasterToEdge.php
class FreshRSS_Migration_2021_04_24_GitCheckoutFromMasterToEdge {
	public static function migrate() {
		if (!is_dir(FRESHRSS_PATH . '/.git/')) {
			return true; // not a Git installation, nothing to do
		}

		exec('git branch --show-current', $output, $return);
		if ($return != 0) {
			throw new Exception('Can’t checkout to edge branch, please change branch manually.');
		}

		$line = is_array($output) ? implode('', $output) : $output;
		if ($line !== 'master' && $line !== 'dev') {
			return true; // not on master or dev, nothing to do
		}

		unset($output);
		exec('git checkout edge --guess -f', $output, $return);
		if ($return != 0) {
			throw new Exception('Can’t checkout to edge branch, please change branch manually.');
		}

		unset($output);
		exec('git reset --hard FETCH_HEAD', $output, $return);
		if ($return != 0) {
			throw new Exception('Can’t checkout to edge branch, please change branch manually.');
		}

		return true;
	}
}
```

Reply:
Here you are https://github.com/FreshRSS/FreshRSS/pull/3589/commits/e09a1e9a1558c4cbfd62cd3f7032ef9cb917f7b3

Reply:
@marienfressinaud It does not look like the migration errors / results / messages are visible in FreshRSS logs though, only in PHP errors (Web server logs).

Reply:
I also have a case of white page when using Migration